src/index.d.ts

```diff
@@ -0,0 +1,1373 @@
+}
+
+export interface NavigationContainerComponent
+  extends React.Component<
+  NavigationContainerProps & NavigationNavigatorProps<any>
+  > {
+  dispatch: NavigationDispatch;
+}
+
+export interface NavigationContainer
+  extends React.ComponentClass<
+  NavigationContainerProps & NavigationNavigatorProps<any>
+  > {
+  new(
+    props: NavigationContainerProps & NavigationNavigatorProps<any>,
+    context?: any
+  ): NavigationContainerComponent;
+
+  router: NavigationRouter<any, any>;
+  screenProps: { [key: string]: any };
+  navigationOptions: any;
+  state: { nav: NavigationState | null };
+}
+
+export interface StackNavigatorConfig
+  extends NavigationStackViewConfig,
+  NavigationStackRouterConfig {
+  containerOptions?: any;
+}
+
+// Return createNavigationContainer
+export function StackNavigator(
+  routeConfigMap: NavigationRouteConfigMap,
+  stackConfig?: StackNavigatorConfig
+): NavigationContainer;
+
+export function createStackNavigator(
+  routeConfigMap: NavigationRouteConfigMap,
+  stackConfig?: StackNavigatorConfig
+): NavigationContainer;
+
+export interface SwitchNavigatorConfig {
+  initialRouteName: string;
+  resetOnBlur?: boolean;
+  paths?: NavigationPathsConfig;
+  backBehavior?: 'none' | 'initialRoute';
+}
+
+// Return createNavigationContainer
+export type _SwitchNavigatorConfig = NavigationSwitchRouterConfig;
+
+export function SwitchNavigator(
+  routeConfigMap: NavigationRouteConfigMap,
+  switchConfig?: SwitchNavigatorConfig
+): NavigationContainer;
+
+export function createSwitchNavigator(
+  routeConfigMap: NavigationRouteConfigMap,
+  switchConfig?: SwitchNavigatorConfig
+): NavigationContainer;
+
+// DrawerItems
+export const DrawerItems: React.ComponentType<DrawerItemsProps>;
+
+export interface DrawerItemsProps {
+  navigation: NavigationScreenProp<DrawerNavigationState>;
+  items: NavigationRoute[];
+  activeItemKey?: string;
+  activeTintColor?: string;
+  activeBackgroundColor?: string;
+  inactiveTintColor?: string;
+  inactiveBackgroundColor?: string;
+  getLabel: (scene: DrawerScene) => React.ReactNode | string;
+  renderIcon: (scene: DrawerScene) => React.ReactNode;
+  onItemPress: (info: DrawerItem) => void;
+  itemsContainerStyle?: StyleProp<ViewStyle>;
+  itemStyle?: StyleProp<ViewStyle>;
+  labelStyle?: StyleProp<TextStyle>;
+  activeLabelStyle?: StyleProp<TextStyle>;
+  inactiveLabelStyle?: StyleProp<TextStyle>;
+  iconContainerStyle?: StyleProp<ViewStyle>;
+  drawerPosition: 'left' | 'right';
+}
+export interface DrawerScene {
+  route: NavigationRoute;
+  focused: boolean;
+  index: number;
+  tintColor?: string;
+}
+export interface DrawerItem {
+  route: NavigationRoute;
+  focused: boolean;
+}
+
+/**
+ * Drawer Navigator
+ */
+export interface DrawerViewConfig {
+  drawerBackgroundColor?: string;
+  drawerWidth?: number;
+  drawerPosition?: 'left' | 'right';
+  contentComponent?: React.ComponentType<DrawerItemsProps>;
+  contentOptions?: any;
+  style?: StyleProp<ViewStyle>;
```

Comment:
`drawerType` and `overlayColor` are missing here, and `drawerWidth` also accepts a function;

```ts
drawerType?: 'front' | 'back' | 'slide';
overlayColor?: string;
drawerWidth?: number | (() => number);
```